src/alda/parser.clj

```diff
@@ -20,16 +20,12 @@
           :slur              (constantly :slur)
           :flat              (constantly :flat)
           :sharp             (constantly :sharp)
+          :natural           (constantly :natural)
           :dots              #(hash-map :dots (count %))
           :note-length       #(list* 'alda.lisp/note-length %&)
           :duration          #(list* 'alda.lisp/duration %&)
-          :pitch             (fn [s]
-                               (list* 'alda.lisp/pitch
-                                      (keyword (str (first s)))
-                                      (map #(case %
-                                              \- :flat
-                                              \+ :sharp)
-                                           (rest s))))
+          :pitch             #(list* 'alda.lisp/pitch %&)
+          :pitch-notation    #(list* 'alda.lisp/parse-pitch %)
```

Comment:
The reason that I added :pitch-notation as a first class member of the parse is that I needed it in the key signature as well. If I were to take this approach, wouldn't I also have to parse the :pitch-notation in parse-key-signature()? I could certainly do that, but it feels like a lot of parse action happening to far along in the process. I may be misunderstand the flow though.